src/message.rs

```diff
@@ -760,6 +761,16 @@ impl Message {
             .await
             .is_ok()
     }
+
+    /// Gets the error of the message.
+    ///
+    /// Possible errors:
+    /// * "No valid signature"
+    /// * "Decryption failed"
+    /// * a non delivery notification
```

Comment:
This has been a long discussion and many things have been learned about this API in this PR.  I don't feel like this fairly represents what we know, so here my attempt:

```suggestion
    /// Gets the error status of the message.
    ///
    /// A message can have an associated error status if something went wrong when sending or
    /// receiving message itself.  The error status is free-form text and should not be further parsed,
    /// rather it's presence is meant to indicate *something* went wrong with the message and the
    /// text of the error is detailed information on what.
    ///
    /// Some common reasons error can be associated with messages are:
    /// * Lack of valid signature on an e2ee message, usually for received messages.
    /// * Failure to decrypt an e2ee message, usually for received messages.
    /// * When a message could not be delivered to one or more recipients the non-delivery
    ///    notification text can be stored in the error status.
```

Please feel free to further improve and tweak, as i said this has been a long discussion and I almost certainly missed some important subtleties.